lib/_http_common.js

```diff
@@ -43,22 +23,11 @@ function parserOnHeadersComplete(versionMajor, versionMinor, headers, method,
                                  url, statusCode, statusMessage, upgrade,
                                  shouldKeepAlive) {
   var parser = this;
-
-  if (!headers) {
-    headers = parser._headers;
-    parser._headers = [];
-  }
-
-  if (!url) {
-    url = parser._url;
-    parser._url = '';
-  }
-
-  parser.incoming = new IncomingMessage(parser.socket);
-  parser.incoming.httpVersionMajor = versionMajor;
-  parser.incoming.httpVersionMinor = versionMinor;
-  parser.incoming.httpVersion = versionMajor + '.' + versionMinor;
-  parser.incoming.url = url;
+  var stream = parser.incoming = new IncomingMessage(parser.socket);
```

Comment:
FWIW, I tried this optimization when trying to optimize the same method and I saw a decrease in perf ([see n=1, before, vs n=2, with change](http://s29.postimg.org/x097bc3cn/graph.png))